docs/source/reference/training.rst

```diff
@@ -35,33 +35,93 @@ Updater
 
 .. _extensions:
 
-Extension
----------
+Extensions
+----------
+
+An extension is a callable object that can perform arbitrary actions during the training loop.
+Extensions can be registered to :class:`Trainer` by using :func:`Trainer.extend` method, and they are invoked when the :ref:`Trigger <triggers>` condition is satisfied.
+
+In addition to the built-in extensions listed below, you can define your own extension using :class:`Extension` or :meth:`make_extension` decorator.
```

Comment:
How about this small change `... own extension by implementing Extension or using the make_extension() decorator`?